test/integration/packwerk/load_test.rb

```diff
@@ -0,0 +1,34 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+require "active_support/testing/isolation"
+
+module Packwerk
+  class LoadTest < Minitest::Test
+    include ActiveSupport::Testing::Isolation
+    GEM_LIB = File.expand_path("../../../lib", __dir__)
+
+    setup { unload_packwerk }
+
+    test "does not autoload early" do
+      require "packwerk"
+
+      assert_equal(
+        ["#{GEM_LIB}/packwerk.rb", "#{GEM_LIB}/packwerk/version.rb"].sort,
+        loaded_packwerk_files.sort,
+      )
+    end
+
+    private
+
+    def loaded_packwerk_files
+      $LOADED_FEATURES.select { |path| path.starts_with?(GEM_LIB) }
+    end
+
+    def unload_packwerk
```

Comment:
I'm not super happy about this hack. Is there a way to test in isolation without needing to undo what the test runner process has already done?

Reply:
Great idea! I'll give that a shot.

Reply:
I initially had this in integration, but our integration tests are just interacting with Rails, and this seemed different, so I made a `test:loading` suite with a different helper file. Seems to work, and this would be a good place to put https://github.com/Shopify/packwerk/pull/334 too.